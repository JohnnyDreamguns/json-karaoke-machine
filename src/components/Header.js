import React from 'react';
import { Link } from 'react-router-dom';
import NoSoundModal from './NoSoundModal';

const Header = ({ play, isPlaying, tempo, handleChangeTempo }) => (
  <nav className="navbar navbar-expand-lg navbar-light fixed-top" id="mainNav">
    <div className="container">
      <nav>
        <Link to="/" className="navbar-brand js-scroll-trigger">
          JSON Karaoke Machine
        </Link>
        <span>|</span>
        <Link to="/about" className="navbar-brand js-scroll-trigger">
          About
        </Link>
      </nav>

      {play && (
        <div className="play-controls">
          {isPlaying ? <NoSoundModal /> : ''}
          <button
            className="btn btn-outline btn-xl"
            data-playing="false"
            role="switch"
            aria-checked="false"
            onClick={play}
          >
            {isPlaying ? 'Stop' : 'Play!'}
          </button>

          <div className="tempo-control">
            <label forhtml="tempo">{tempo}bpm</label>
            <input
              type="range"
              id="tempo"
              min="70"
              max="250"
              value={tempo}
              step="1"
              onChange={handleChangeTempo}
            />
          </div>
        </div>
      )}
    </div>
  </nav>
);

export default Header;
