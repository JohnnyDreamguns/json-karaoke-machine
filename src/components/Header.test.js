import 'babel-polyfill';
import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

const defaultProps = {
  play: jest.fn(),
  isPlaying: false,
  tempo: 120,
  handleChangeTempo: jest.fn()
};

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header {...defaultProps} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
