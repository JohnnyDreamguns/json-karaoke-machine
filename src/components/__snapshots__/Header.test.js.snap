// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light fixed-top"
  id="mainNav"
>
  <div
    className="container"
  >
    <nav>
      <a
        className="navbar-brand js-scroll-trigger"
        href="/"
        onClick={[Function]}
      >
        JSON Karaoke Machine
      </a>
      <span>
        |
      </span>
      <a
        className="navbar-brand js-scroll-trigger"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </nav>
    <div
      className="play-controls"
    >
      
      <button
        aria-checked="false"
        className="btn btn-outline btn-xl"
        data-playing="false"
        onClick={[MockFunction]}
        role="switch"
      >
        Play!
      </button>
      <div
        className="tempo-control"
      >
        <label
          forhtml="tempo"
        >
          120
          bpm
        </label>
        <input
          id="tempo"
          max="250"
          min="70"
          onChange={[MockFunction]}
          step="1"
          type="range"
          value={120}
        />
      </div>
    </div>
  </div>
</nav>
`;
